webapp/channels/src/packages/mattermost-redux/src/selectors/entities/users.ts

```diff
@@ -824,3 +834,15 @@ export const getLastActiveTimestampUnits: (state: GlobalState, userId: string) =
         return timestampUnits;
     },
 );
+
+export function getFirstDayOfWeekForCurrentUser(state: GlobalState): number {
+    return Number(getCurrentUser(state)?.props.first_day_of_week ?? '0');
+}
+
+export const getFirstDayOfWeekLabel: (state: GlobalState) => string = createSelector(
```

Comment:
I like the idea here, but I don't think we're going to be able to keep these here since we need to be able to translate them into other languages which can't be done from code inside the `mattermost-redux` folder. Maybe instead of having this and `daysOfWeek` located here, you could instead make a `getDayOfWeekLabel` function in `utils/datetime` that takes a number and returns the translated string using `localizeMessage`?

That's not perfect since it uses `localizeMessage` but given we're needing to use this from a class component, we can't use `useIntl` to get the translations easier